### ISSUE TITLE
[muongun] cubature memory leaks (Trac #1471)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1463]
cubature.c

line 1015:
http://software.icecube.wisc.edu/static_analysis/2015-12-07-030322-70644-1/report-eb2bb2.html#EndPath

Also, line 969 (the realloc) looks fishy. If it fails it will return a NULL pointer, but the previous pointer is still valid memory that should be deallocated.  This code seems to let that previous memory leak when it overwrites the pointer address.

Migrated from https://code.icecube.wisc.edu/ticket/1471
```json
{
    "status": "closed", 
    "changetime": "2016-03-18T21:14:10", 
    "description": "cubature.c\n\nline 1015:\nhttp://software.icecube.wisc.edu/static_analysis/2015-12-07-030322-70644-1/report-eb2bb2.html#EndPath\n\nAlso, line 969 (the realloc) looks fishy. If it fails it will return a NULL pointer, but the previous pointer is still valid memory that should be deallocated.  This code seems to let that previous memory leak when it overwrites the pointer address.", 
    "reporter": "david.schultz", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1458335650323600", 
    "component": "combo simulation", 
    "summary": "[muongun] cubature memory leaks", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2015-12-07T22:35:06", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151214T21:05:55: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 140433:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="140433"
> Be more dubious of the return value of realloc(). Fixes #1463.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:10: 
* olivas commented:

> Milestone offline V15-12-00 deleted


* olivas changed milestone from "offline V15-12-00" to ""
